tools/changeDelayTimelock/changeDelayTimelock.ts

```diff
@@ -0,0 +1,176 @@
+                async function overrideFeeData() {
+                    const feedata = await ethers.provider.getFeeData();
+                    return new ethers.FeeData(
+                        null,
+                        ((feedata.maxFeePerGas as bigint) * BigInt(parameters.multiplierGas)) / 1000n,
+                        ((feedata.maxPriorityFeePerGas as bigint) * BigInt(parameters.multiplierGas)) / 1000n
+                    );
+                }
+                currentProvider.getFeeData = overrideFeeData;
+            }
+        }
+    }
+
+    // Load deployer
+    let deployer;
+    if (parameters.deployerPvtKey) {
+        deployer = new ethers.Wallet(parameters.deployerPvtKey, currentProvider);
+    } else if (process.env.MNEMONIC) {
+        deployer = ethers.HDNodeWallet.fromMnemonic(
+            ethers.Mnemonic.fromPhrase(process.env.MNEMONIC),
+            "m/44'/60'/0'/0/0"
+        ).connect(currentProvider);
+    } else {
+        [deployer] = await ethers.getSigners();
+    }
+    
+
+    const timelockContractFactory = await ethers.getContractFactory("PolygonZkEVMTimelock", deployer);
+    const timelockContract = (await timelockContractFactory.attach(
+        parameters.timelockContractAddress
+    )) as PolygonZkEVMTimelock;
+
+    console.log("#######################\n");
+    console.log("timelockContract address: ", timelockContract.target)
+    console.log("#######################\n");
+
+    const timelockDelay = parameters.timeLockDelay ? parameters.timeLockDelay : Number(await timelockContract.getMinDelay());
+    const salt = parameters.timelockSalt || ethers.ZeroHash;
+    const predecessor = parameters.predecessor || ethers.ZeroHash;
+
+    const operation = genOperation(
+        parameters.timelockContractAddress,
+        0, // value
+        timelockContract.interface.encodeFunctionData(
+            'updateDelay',
+            [parameters.newMinDelay],
+        ),
+        predecessor, // predecessor
+        salt // salt
+    );
+
+    // Schedule operation
+    const scheduleData = timelockContractFactory.interface.encodeFunctionData("schedule", [
+        operation.target,
+        operation.value,
+        operation.data,
+        operation.predecessor,
+        operation.salt,
+        timelockDelay,
+    ]);
+
+    // Execute operation
+    const executeData = timelockContractFactory.interface.encodeFunctionData("execute", [
+        operation.target,
+        operation.value,
+        operation.data,
+        operation.predecessor,
+        operation.salt,
+    ]);
+
+    console.log("timelockDelay: ", timelockDelay)
+    console.log({scheduleData});
+    console.log({executeData});
+   
+    outputJson.scheduleData = scheduleData;
+    outputJson.executeData = executeData;
+    outputJson.minDelay = timelockDelay;
+    outputJson.functionData = {
+        function: 'updateDelay',
+        parameters: parameters.newMinDelay
+    }
+
+    if(parameters.sendSchedule) {
```

Comment:
I would personally not add this functionality to this script.
It will simplify a lot the input parameters and it does make sense for long timelock delays.

Usually, timelock schedule/executions can be done in different ways: metamask (hex data), cast or it could be a `.ts` tool.
This code could be converted into another tool, such like `sendTimelockData` that would be agnostic to the data sent and it could be used for any scheduled/executed timelock data.

Also, removing the ability to send the transaction aligns with the other tools that also creates timelock data